httpx/concurrency.py

```diff
@@ -108,6 +108,9 @@ def __init__(
 
         return data
 
+    def is_connection_dropped(self) -> bool:
+        return self.stream_reader.at_eof()
```

Comment:
Probably should clarify this seemingly simplistic implementation. The `StreamReaderProtocol.connection_lost` callback [calls `stream_reader.feed_eof`](https://github.com/python/cpython/blob/master/Lib/asyncio/streams.py#L651) which in turn [sets the EOF flag](https://github.com/python/cpython/blob/master/Lib/asyncio/streams.py#L827) which `at_eof` returns.